homeassistant/components/smsapi/notify.py

```diff
@@ -0,0 +1,42 @@
+"""Support for the SMSApi platform."""
+import logging
+
+from smsapi.client import SmsApiPlClient
+from smsapi.exception import SmsApiException
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    PLATFORM_SCHEMA, BaseNotificationService)
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_RECIPIENT
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Required(CONF_RECIPIENT): cv.string,
+})
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the SMSApi notification service."""
+    return SMSApiNotificationService(
+        config[CONF_ACCESS_TOKEN], config[CONF_RECIPIENT])
+
+
+class SMSApiNotificationService(BaseNotificationService):
+    """Implement a notification service for the SMSApi service."""
+
+    def __init__(self, access_token, recipient):
+        """Initialize the service."""
+
+        self.smsapi = SmsApiPlClient(access_token=access_token)
```

Comment:
Can we test that the token is correct already during setup?

Reply:
We can use some basic method like fetching the balance. If you point me to some other, similar component that tests setup, I can try to implement it. I checked several of them but didn't find any suitable one.

Reply:
Instantiate the client first. Test the token. If the token is bad, log error and return `None` in `get_service`. If the token is good, pass the client to the `SMSApiNotificationService` when instantiating that.